projects/plugins/boost/app/modules/optimizations/lcp/class-lcp-optimize-bg-image.php

```diff
@@ -0,0 +1,188 @@
+				esc_url( Image_CDN_Core::cdn_url( $breakpoint['base_image'] ) ),
+				esc_attr( $breakpoint['media_query'] ),
+				esc_attr( $image_set_string )
+			);
+		}
+	}
+
+	public function add_bg_style_override() {
+		$selectors = array();
+
+		foreach ( $this->lcp_data as $lcp_data ) {
+			if ( in_array( $lcp_data['element'], $selectors, true ) ) {
+				// If we already printed the styling for this element, skip it.
+				continue;
+			}
+			$selectors[] = $lcp_data['element'];
+
+			$lcp_optimizer = new LCP_Optimization_Util( $lcp_data );
+			$image_url     = $lcp_optimizer->get_image_to_preload();
+			if ( empty( $image_url ) ) {
+				continue;
+			}
+
+			$styles                 = array();
+			$responsive_image_rules = $this->get_responsive_image_rules( $lcp_data );
+
+			// Add responsive image styling.
+			foreach ( $responsive_image_rules as $breakpoint ) {
+				$image_set = array();
+				foreach ( $breakpoint['image_set'] as $image ) {
+					$image_set[] = sprintf( 'url(%s) %sx', $image['url'], $image['dpr'] );
+				}
+
+				$image_set_string = implode( ', ', $image_set );
+
+				$styles[] = sprintf(
+					'@media %1$s { %2$s { background-image: url(%3$s) !important; background-image: -webkit-image-set(%4$s) !important; background-image: image-set(%4$s) !important; } }',
+					$breakpoint['media_query'],
+					$lcp_data['element'],
+					$breakpoint['base_image'],
+					$image_set_string
+				);
+			}
+
+			$bg_styling = PHP_EOL . '<style id="jetpack-boost-lcp-background-image">' . PHP_EOL;
+			// Ensure no </style> tag (or any HTML tags) in output.
+			$bg_styling .= wp_strip_all_tags( implode( PHP_EOL, $styles ) ) . PHP_EOL;
+			$bg_styling .= '</style>' . PHP_EOL;
+
+			// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+			echo $bg_styling;
+		}
+	}
+
+	private function get_responsive_image_rules( $lcp_data ) {
+		if ( empty( $lcp_data['breakpoints'] ) ) {
+			return array();
+		}
+
+		$lcp_optimizer = new LCP_Optimization_Util( $lcp_data );
+		$image_url     = $lcp_optimizer->get_image_to_preload();
+
+		if ( empty( $image_url ) ) {
+			return array();
+		}
+
+		$styles = array();
+		// Reverse the array to go from smallest to largest.
+		foreach ( array_reverse( $lcp_data['breakpoints'] ) as $breakpoint ) {
+			if ( empty( $breakpoint ) ) {
+				continue;
+			}
+			if ( ! isset( $breakpoint['widthValue'] ) ) {
+				continue;
+			}
+
+			// The Cloud should always return a fixed pixel width for background images, so catering for that is easy peasy.
+			if ( ! isset( $breakpoint['imageWidths'][0] ) ) {
+				continue;
+			}
+
+			$image_width = $breakpoint['imageWidths'][0];
+
+			$media_query = array();
+			if ( isset( $breakpoint['minWidth'] ) ) {
+				$media_query[] = sprintf( '(min-width: %spx)', $breakpoint['minWidth'] );
+			}
+			if ( isset( $breakpoint['maxWidth'] ) ) {
+				$media_query[] = sprintf( '(max-width: %spx)', $breakpoint['maxWidth'] );
+			}
```

Comment:
I think there is a situation where neither minWidth nor maxWidth will be defined. Example: if the LCP image is consistently 300px [as in this page](https://classy-chaja-a151d6.netlify.app/test-page.html). This will result in invalid CSS like:

```css
@media { ... }
```

I think a quick fix would be to put `all` as a value if the array is empty.